projects/packages/connection/src/class-rest-connector.php

```diff
@@ -521,4 +544,59 @@ public function connection_authorize_url( $request ) {
 		);
 	}
 
+	/**
+	 * The endpoint tried to partially or fully reconnect the website to WP.com.
+	 *
+	 * @since 9.9.0
+	 *
+	 * @param \WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return \WP_REST_Response|WP_Error
+	 */
+	public static function update_user_token( $request ) {
+		$token_parts = explode( '.', $request['user_token'] );
+
+		if ( count( $token_parts ) !== 3 || ! (int) $token_parts[2] || ! ctype_digit( $token_parts[2] ) ) {
+			return new WP_Error( 'invalid_argument_user_token', esc_html__( 'Invalid user token is provided', 'jetpack' ) );
+		}
+
+		$user_id = (int) $token_parts[2];
+
+		if ( false === get_userdata( $user_id ) ) {
+			return new WP_Error( 'invalid_argument_user_id', esc_html__( 'Invalid user id is provided', 'jetpack' ) );
+		}
+
+		$connection = new Manager();
+
+		if ( ! $connection->is_connected() ) {
+			return new WP_Error( 'site_not_connected', esc_html__( 'Site is not connected', 'jetpack' ) );
+		}
+
+		$is_connection_owner = isset( $request['is_connection_owner'] )
+			? (bool) $request['is_connection_owner']
+			: ( new Manager() )->get_connection_owner_id() === $user_id;
+
+		( new Tokens() )->update_user_token( $user_id, $request['user_token'], $is_connection_owner );
+
+		do_action( 'jetpack_update_user_token_success', $user_id, $request['user_token'] );
```

Comment:
Hey @sergey!

Since this action fires **after** we updated the user token, maybe we could rename it to : `jetpack_updated_user_token` (to be more consistent with other actions like `jetpack_unlinked_user`, `jetpack_site_registered` etc)?

Also can we pls add a docblock here for our new shiny hook?

Then we may go ahead and add it in `Error_Handler`:
```
add_action( 'jetpack_updated_user_token', array( $this, 'delete_all_errors' ) );
```

Reply:
Done: https://github.com/Automattic/jetpack/pull/19994/commits/4089b2015de3d3fe202a3c1f00b5bf0b8e394d83